### PR TITLE
Move junit property fixtures and reporting hook

### PR DESCRIPTION
**Pull request by mshriver**
This was in conftest, which was getting copied for the unit test.

Since the file with the hooks/fixtures needs to be copied during the
unit test execution, split the reporting fixtures into their own file
for more specific unit testing.

This allows for content in foreman/conftest.py to not have to be unit
test safe, when the module is not intended for use in unit tests

### CODE REVIEW

**Comment by mshriver** [comment]
trigger: test-robottelo

File: pytest_fixtures/reporting_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def pytest_report_header(config):
+    """Called when pytest session starts"""
+    messages = []
+
+    shared_function_enabled = 'OFF'
+    scope = ''
+    storage = 'file'
+    if setting_is_set('shared_function'):
```
Comment (JacobCallahan) [inline pytest_fixtures/reporting_fixtures.py:21]: is there a case in which this wouldn't be set? I believe the facade layer runs through all settings before this point.

File: pytest_fixtures/reporting_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import pytest
+from _pytest.junitxml import xml_key
+from xdist import get_xdist_worker_id
+
+from robottelo.config import settings
+from robottelo.decorators import setting_is_set
+
+
+FMT_XUNIT_TIME = "%Y-%m-%dT%H:%M:%S"
```
Comment (JacobCallahan) [inline pytest_fixtures/reporting_fixtures.py:11]: single quotes :)

File: pytest_fixtures/reporting_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    return messages
+
+
+def pytest_sessionstart(session):
+    """Workaround for https://github.com/pytest-dev/pytest/issues/7767
+
+    remove if resolved and set autouse=True for record_testsuite_timestamp_xml fixture
+    """
+    if get_xdist_worker_id(session) == 'master':
+        if session.config.pluginmanager.hasplugin("junitxml"):
```
Comment (JacobCallahan) [inline pytest_fixtures/reporting_fixtures.py:41]: single quotes :)
other instances in this file as well

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
small things

File: pytest_fixtures/reporting_fixtures.py (hunk 1 above)
Comment (mshriver) [reply pytest_fixtures/reporting_fixtures.py:21]: This is unchanged from the existing functionality, I just moved the function.  I can update it, but would prefer to keep the scope related to how the unit test operates and whether the hook runs on xdist workers.

File: pytest_fixtures/reporting_fixtures.py (hunk 3 above)
Comment (mshriver) [reply pytest_fixtures/reporting_fixtures.py]: replaced quotes in this module and the unit test module.

**Review by JacobCallahan — APPROVED** [review]
ACK

**Comment by JacobCallahan** [comment]
@rplevka can you check this out? Thanks!

**Review by rplevka — APPROVED** [review]

**Pull request merged**